app/helpers/submission_helper.rb

```diff
@@ -6,4 +6,16 @@ def plus_fix(f)
       sprintf("%.2f", f.round(2))
     end
   end
+
+  def released(assessment, submission)
```

Comment:
Easier/better to do `submission.scores.pluck(:released).all?`

So that it's considered released as long as all problems with scores are released.

e.g. the button will show "unreleased grade" despite the nil manual score

![Screenshot 2024-02-19 at 22 01 58](https://github.com/autolab/Autolab/assets/9074856/0026330e-709a-44b6-835e-df97c87e67bc)

Reply:
But note that this would become pretty similar to `grades_released?` in `submission.rb`. Maybe put the methods together?

```ruby
  def problems_released?
    scores.pluck(:released).all?
  end

  def grades_released?(as_seen_by)
    include_unreleased = as_seen_by.course_assistant? || as_seen_by.instructor?
    problems_released? || include_unreleased
  end
```